### DISCUSSION TITLE
Are there any plans to offer the eslint plugin?

### DISCUSSION THREAD

**Discussion by shinGangan** [nuxt/content#2568] [Ideas]
Nuxt Content has Markdown Components notation.
I am very grateful to [MDC VS Code extension](https://marketplace.visualstudio.com/items?itemName=Nuxt.mdc&ssr=false#overview) for its help in implementing Nuxt Content.

However, there are many cases where it would be nice to be able to check statically like eslint-plugin-vue.
For example.

- I want to detect a single `:` missing in an inline component before merging pull request
- If the content creator is a non-engineer or does not use VS Code

etc.

### Disucuss

We believe the segregation will be as follows.

`eslint-plugin-vue` → static check of `/components/content/*`
`eslint-plugin-nuxt-content (tentative)` → static check of `/content/*`

However, we think it is necessary to consider the segregation from [MarkdownLint](https://github.com/DavidAnson/markdownlint).
(e.g., write **No MarkdownLint** in Docs like **No Prettier** in [the Nuxt Contribution Guide](https://nuxt.com/docs/community/contribution#no-prettier)).

**Comment by JessicaSachs** [comment]
👀 Have there been any recent musings about what to do wrt linting? I'm seeing this [MonarchMDC](https://github.com/nuxtlabs/monarch-mdc) Monaco language tool repo from the NuxtLabs team is very active, so I'm wondering what's cooking!

## Current Configuration Workaround

Here's my `.markdownlint.mjs` configuration right now:

```ts
// @ts-check

export default {
  config: {
    // Gets confused with MDC syntax
    MD003: false, // YAML front matter uses `---`
    MD007: false, // Indentation used for MDC (lists)
    MD018: false, // Slots are not headers `#title` or `#description`
    MD022: false, // Thinks YAML is that headers should be surrounded by blank lines
    MD023: false, // Indentation used for MDC
  },
  ignores: [
    '**/*.code-search',
    '**/bower_components',
    '**/node_modules',
    '**/.git',
    '**/vendor',
  ],
  globs: ['./content/**/*.md'],
}
```

## Needs

1. I'd really like to use a formatter to indent blocks consistently
2. I'd like to enforce linking styles within `content`.
	1. In particular, we're struggling with relative linking including the file-based numbering `./02.folder/file.md` may change without the intention of breaking the link.
3. I'd like to integrate with any reasonable linter workflow. Eslint-controlled prettier, markdownlint... I don't care 😸

### Nice-to-have

The ability to write custom rules in any of the major formatter/linter ecosystems so that I can enforce props, style, etc.
